### ISSUE TITLE
Upgrade to Next.js 14+

### ISSUE THREAD

**Issue by jayantbh** [middlewarehq/middleware#597]
## Why do we need this ?

This is a regular updates and hygiene thing.
Next.js 14 offers various performance and tooling improvements that we could benefit from.

The main thing to keep in mind for this one is that pulling this off will need thorough and comprehensive testing of the whole app, and builds.

Upgrading Next.js should include updating any related packages (such as React, etc.), and any resulting changes to types or how third party packages/libs/functions are used.

Basically, nothing should really much different for any dev or user when they use Middleware after this issue is resolved.

**Comment by jayantbh** [comment]
It's certainly available. But I'd recommend tackling the axios one first before taking this on, since this is certainly a bigger issue and the other one will help get you started while keeping this free in case someone else is able to give it a shot first.

**Comment by jayantbh** [comment]
@Reterics this is open for you now, if interested. :)
Thanks for your other recent contribution. 🚀

**Comment by Reterics** [comment]
> @Reterics this is open for you now, if interested. :) Thanks for your other recent contribution. 🚀

Thank you, I am happy to work on it :)
